tests/src/ut_test_kvp.c

```diff
@@ -199,37 +226,49 @@ void test_ut_kvp_open_malloced( void )
     ut_kvp_close(pInstance);
 
     /* Negative Read Test, NULL PARAM */
-    UT_LOG_STEP("ut_kvp_open( pInstance, NULL ) - Negative");
-    status = ut_kvp_open( pInstance, NULL, true );
+    UT_LOG_STEP("ut_kvp_openMemory( pInstance, NULL ) - Negative");
+    status = ut_kvp_openMemory( pInstance, NULL, -1);
     UT_ASSERT( status == UT_KVP_STATUS_INVALID_PARAM );
 
     /* data doesn't exist */
-    UT_LOG_STEP("ut_kvp_open( pInstance, %s - filename doesn't exist ) - Negative", KVP_VALID_TEST_NO_FILE);
+    UT_LOG_STEP("ut_kvp_openMemory( pInstance, %s - filename doesn't exist ) - Negative", KVP_VALID_TEST_NO_FILE);
     const char *kvp_str = "  ";
     char* kvp_buffer = strdup(kvp_str);
-    if (!kvp_buffer) {
-        UT_LOG_ERROR("strdup");
+    if (kvp_buffer)
+    {
+        status = ut_kvp_openMemory(pInstance, kvp_buffer, -1);
+        UT_ASSERT(status == UT_KVP_STATUS_PARSING_ERROR);
```

Comment:
why is this a PARSING_ERROR , should be invalid param?

Reply:
We return invalid param when 2nd param is NULL
```
if (pData == NULL)
    {
        assert( pData != NULL );
        UT_LOG_ERROR( "Invalid Param [string]" );
        return UT_KVP_STATUS_INVALID_PARAM;
    }
```
In this case its not NULL. But fails during the libfyaml library call.

Reply:
This file is now moved to ut-control